polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -84,14 +88,52 @@ bool isCandidate(LoopLikeOpInterface loop) {
   return true;
 }
 
+unsigned getLocalMemRemain(Operation *module) {
+  assert(module->hasTrait<OpTrait::SymbolTable>() &&
+         "Expecting module with SymbolTable trait");
+  unsigned localMemRemain = LocalMemSize;
+  module->walk([&localMemRemain](memref::GlobalOp global) {
+    auto MT = global.getType();
+    auto MS = dyn_cast_or_null<sycl::AccessAddrSpaceAttr>(MT.getMemorySpace());
+    if (!MS || MS.getValue() != sycl::AccessAddrSpace::LocalAccess)
+      return WalkResult::advance();
+    unsigned globalSize = MT.getElementTypeBitWidth() * MT.getNumElements() / 8;
+    if (globalSize >= localMemRemain) {
+      localMemRemain = 0;
+      return WalkResult::interrupt();
+    }
+    localMemRemain = localMemRemain - globalSize;
+    return WalkResult::advance();
+  });
+  return localMemRemain;
+}
+
+memref::GlobalOp getWGLocalMem(LoopLikeOpInterface loop) {
```

Comment:
Add a comment to the utility function. (///)

Reply:
done.